auto-update-dataset/python/prStatusCheck.py

```diff
@@ -0,0 +1,237 @@
+import requests
+import re
+import time
+import logging
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+# Setup logging
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s',
+    handlers=[
+        logging.StreamHandler(sys.stdout),
+        logging.FileHandler("pr-status-update.log")
+    ]
+)
+logger = logging.getLogger(__name__)
+
+# Try to read token from project dir
+script_dir = os.path.dirname(os.path.abspath(__file__))
+token_file = os.path.join(script_dir, "token.txt")
+github_token = None
+if os.path.exists(token_file):
+    try:
+        with open(token_file, "r") as f:
+            github_token = f.read().strip()
+    except Exception as e:
+        logger.warning(f"Failed to read token file: {e}")
+
+# Set up request headers
+HEADERS = {}
+if github_token:
+    HEADERS["Authorization"] = f"Bearer {github_token}"
+else:
+    logger.warning("github_token not set. Rate limits will be strict.")
+
+# Regular expression to match GitHub URLs
+GITHUB_URL_PATTERN = re.compile(r"github\.com/([^/]+)/([^/]+)/(?:issues|pull)/(\d+)")
+
+# Data Source URLs
+DATA_URLS = {
+    'py-data.csv': "https://raw.githubusercontent.com/TestingResearchIllinois/idoft/main/py-data.csv",
+    'pr-data.csv': "https://raw.githubusercontent.com/TestingResearchIllinois/idoft/main/pr-data.csv",
+    'gr-data.csv': "https://raw.githubusercontent.com/TestingResearchIllinois/idoft/main/gr-data.csv"
+}
+
+# File paths
+IGNORE_PATH = os.path.abspath(os.path.join(script_dir, "../ignore.csv"))
+REPORT_PATH = os.path.join(script_dir, "report.log")
+MANUAL_CHECK_PATH = os.path.join(script_dir, "manual-check.log")
+
+def fetch_status(url, original_status):
+    match = GITHUB_URL_PATTERN.search(url)
+    if not match:
+        logger.warning(f"Invalid GitHub URL format: {url}. Returning original status.")
+        return original_status
+    owner, repo, number = match.groups()
+    pull_url = f"https://api.github.com/repos/{owner}/{repo}/pulls/{number}"
+    try:
+        response = requests.get(pull_url, headers=HEADERS)
+        if response.status_code == 200:
+            data = response.json()
+            state = data.get("state")
+            if state == "open":
+                return "Opened"
+            elif state == "closed":
+                merged = data.get("merged", False)
+                return "Accepted" if merged else "Unknown"
+        elif response.status_code == 404:
+            logger.warning(f"PR {url} returned 404. Manual check recommended. Returning original status.")
+            return original_status
+        else:
+            logger.warning(f"API Error for PR {url}: {response.status_code}. Returning original status.")
+            return original_status
+    except Exception as e:
+        logger.warning(f"Exception fetching PR {url}: {e}. Returning original status.")
+        return original_status
+
+def process_row(filename, index, row):
+    url = row['PR Link']
```

Comment:
Why check for each and every row rather than each and every URL PR? One PR may be in multiple rows, and there's absolutely no need to recheck it multiple times.